crates/cubecl-ir/src/barrier.rs

```diff
@@ -80,6 +89,9 @@ pub enum BarrierOps {
         arrive_count_update: Variable,
         transaction_count_update: Variable,
     },
+    CommitCopyAsync {
+        barrier: Variable,
+    },
```

Comment:
Is there a reason to expose the commit instruction? I guess it's better if we create a batch of copy async calls then commit to create a bigger batch of copy operations that can be optimized by the cuda runtime. If that's the case, I would add documentation on top of the commit function in the frontend telling when to use it and why/how